spec/system/user_spec.rb

```diff
@@ -0,0 +1,149 @@
+# - ユーザ登録のテスト
+#   - ユーザの新規登録ができること
+#   - ユーザがログインしていないのにタスク一覧のページに飛ぼうとした場合ログイン画面に遷移すること
+# - セッション機能のテスト
+#   - ログインができること
+#   - 自分の詳細画面(マイページ)に飛べること
+#   - 一般ユーザが他人の詳細画面に飛ぶとタスク一覧ページに遷移すること
+#   - ログアウトができること
+# - 管理画面のテスト
+#   - 管理者は管理画面にアクセスできること
+#   - 一般ユーザは管理画面にアクセスできないこと
+#   - 管理者はユーザを新規登録できること
+#   - 管理者はユーザの詳細画面にアクセスできること
+#   - 管理者はユーザの編集画面からユーザを編集できること
+#   - 管理者はユーザの削除をできること
+require 'rails_helper'
+RSpec.describe '​ユーザ登録・ログイン・ログアウト機能​・管理画面テスト', type: :system do
+
+  describe '​ユーザ登録のテスト​' do
+    context '​ユーザのデータがなくログインしていない場合​' do
+      it '​ユーザ新規登録のテスト​' do
+        visit new_user_path
+        fill_in 'user_name', with: 'sample'
+        fill_in 'user_email', with: 'sample@example.com'
+        fill_in 'user_password', with: '00000000'
+        fill_in 'user_password_confirmation', with: '00000000'
+        click_on 'sign up'
+        expect(page).to have_content 'sample'
+      end
+      it '​ログインしていない時はログイン画面に飛ぶテスト​' do
+        visit tasks_path
+        expect(current_path).to eq new_session_path
+      end
+    end
+  end
+
+  describe "セッション機能のテスト" do
+    before do
+      @user = FactoryBot.create(:user)
+    end
```

Comment:
1行のみの場合は下記の様な記載が可能です。

また、少し高度なので保留して大丈夫ですが、必ず事前にあるデータを作成する場合は、
`let!(:user) { create(:user) }`と記載しておくことでも対応できます。
`let!`だとbeforeの様に即時実行され、`let`だとletの引数のシンボルが呼ばれた際に実行されます。
```suggestion
    before { @user = FactoryBot.create(:user) }
```